src/PaymentGateways/PayPalCommerce/assets/js/paypal-smart-buttons.js

```diff
@@ -0,0 +1,229 @@
+/* globals jQuery, paypal, Give, FormData */
+import DonationForm from './DonationForm';
+
+/**
+ * PayPal Smart Buttons.
+ */
+class SmartButtons {
+	/**
+	 * Constructor.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param {object} form selector.
+	 */
+	constructor( form ) {
+		this.form = form;
+		this.jQueryForm = jQuery( form );
+		this.ajaxurl = Give.fn.getGlobalVar( 'ajaxurl' );
+	}
+
+	/**
+	 * Render PayPal smart buttons.
+	 *
+	 * @since 2.8.0
+	 */
+	boot() {
+		jQuery( document ).on( 'give_gateway_loaded', { self: this }, this.onGatewayLoadBoot );
+
+		if ( DonationForm.isPayPalCommerceSelected( this.jQueryForm ) ) {
+			this.renderSmartButtons();
+		}
+	}
+
+	/**
+	 * Render paypal buttons when reload payment gateways.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param {object} evt Event object.
+	 * @param {*} response Form fields HTML for gateway.
+	 * @param {string} formIdAttr Form Id attribute value.
+	 */
+	onGatewayLoadBoot( evt, response, formIdAttr ) {
+		const self = evt.data.self;
+		if ( formIdAttr === self.form.getAttribute( 'id' ) && DonationForm.isPayPalCommerceSelected( self.jQueryForm ) ) {
+			self.renderSmartButtons();
+		}
+	}
+
+	/**
+	 * Render smart buttons.
+	 *
+	 * @since 2.8.0
+	 */
+	renderSmartButtons() {
+		const smartButtonContainer = this.form.querySelector( '#give-paypal-commerce-smart-buttons-wrap div' );
+
+		if ( ! smartButtonContainer ) {
+			return;
+		}
+
+		const onInitHandler = this.onInitHandler.bind( this );
+		const onClickHandler = this.onClickHandler.bind( this );
+		const createOrderHandler = this.createOrderHandler.bind( this );
+		const onApproveHandler = this.onApproveHandler.bind( this );
+
+		paypal.Buttons( {
+			onInit: onInitHandler,
+			onClick: onClickHandler,
+			createOrder: createOrderHandler,
+			onApprove: onApproveHandler,
+		} ).render( smartButtonContainer );
+	}
+
+	/**
+	 * On init event handler for smart buttons.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param {object} data PayPal button data.
+	 * @param {object} actions PayPal button actions.
+	 */
+	onInitHandler( data, actions ) { // eslint-disable-line
+		// Keeping this for future reference.
+	}
+
+	/**
+	 * On click event handler for smart buttons.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param {object} data PayPal button data.
+	 * @param {object} actions PayPal button actions.
+	 *
+	 * @return {Promise<unknown>} Return wther or not open PayPal checkout window.
+	 */
+	onClickHandler(data, actions) { // eslint-disable-line
+		const formData = new FormData( this.form );
+
+		formData.delete( 'card_name' );
+		this.resetCreditCardFields();
+
+		if ( ! Give.form.fn.isDonationFormHtml5Valid( this.form, true ) ) {
+			return actions.reject();
+		}
+
+		Give.form.fn.removeErrors( this.jQueryForm );
+
+		return Give.form.fn.isDonorFilledValidData( this.form, formData )
+			.then( res => {
+				if ( 'success' === res ) {
+					return actions.resolve();
+				}
+
+				Give.form.fn.addErrors( this.jQueryForm, res );
+				return actions.reject();
+			} );
```

Comment:
Any reason you don't make use of `async...await` for promises in JS?

Reply:
fixed: https://github.com/impress-org/givewp/compare/ae1fb915985b...f48a3f458e48